raysect/optical/observer/imaging/fieldofview.pyx

```diff
@@ -0,0 +1,224 @@
+        self._pixels = pixels
+        self._update_image_geometry()
+
+    @property
+    def sensitivity(self):
+        return self._sensitivity
+
+    @sensitivity.setter
+    def sensitivity(self, value):
+        if value <= 0:
+            raise ValueError("Sensitivity must be greater than zero.")
+        self._sensitivity = value
+
+    @property
+    def azimuth(self):
+        return self._azimuth
+
+    @property
+    def altitude(self):
+        return self._altitude
+
+    @property
+    def angle_h(self):
+        return self._azimuth
+
+    @property
+    def angle_v(self):
+        return self._altitude
+
+    cpdef np.ndarray solid_angle(self):
+        """
+        Returns solid angles of collection (in steradians) for all pixels.
+        """
+
+        cdef:
+            np.ndarray solid_angle1d
+
+        solid_angle1d = (pi / 180.) * self.azimuth_delta * (np.sin((pi / 180.) * (self._altitude + 0.5 * self.altitude_delta)) -
+                                                            np.sin((pi / 180.) * (self._altitude - 0.5 * self.altitude_delta)))
+
+        return solid_angle1d[None, :] * np.ones(self._azimuth.size)[:, None]
+
+    cdef object _update_image_geometry(self):
+        self.azimuth_delta = self._fov[0] / self._pixels[0]
+        self.altitude_delta = self._fov[1] / self._pixels[1]
+
+        self._azimuth = np.linspace(-0.5 * (self._fov[0] - self.azimuth_delta),
+                                    0.5 * (self._fov[0] - self.azimuth_delta),
+                                    self._pixels[0])
+        self._altitude = np.linspace(0.5 * (self._fov[1] - self.altitude_delta),
+                                     -0.5 * (self._fov[1] - self.altitude_delta),
+                                     self._pixels[1])
+
+        self.point_sampler = RectangleSampler3D(self.azimuth_delta * pi / 180., self.altitude_delta * pi / 180.)
+
+    cpdef list _generate_rays(self, int x, int y, Ray template, int ray_count):
+
+        cdef:
+            double azimuth_c, altitude_c, azimuth_p, altitude_p
+            list points, rays
+            Point3D origin
+            Vector3D direction
+            Ray ray
+
+        azimuth_c = self._azimuth[x] * pi / 180.
+        altitude_c = self._altitude[y] * pi / 180.
+        points = self.point_sampler(ray_count)
+
+        # assemble rays
+        rays = []
+        for point in points:
+            azimuth_p = azimuth_c + point.x
+            altitude_p = altitude_c + point.y
+
+            origin = new_point3d(0, 0, 0)
+            direction = new_vector3d(
+                -sin(azimuth_p) * cos(altitude_p),
+                sin(altitude_p),
+                cos(azimuth_p) * cos(altitude_p)
+            ).normalise()
+
+            ray = template.copy(origin, direction)
+
+            rays.append((ray, 1.0))
+
+        return rays
+
+    cpdef double _pixel_sensitivity(self, int x, int y):
+        return self._sensitivity
```

Comment:
There is a finite solid angle, that varies per pixel. The solid angle should be included in this calculation. the sensitivity then becomes an effective area.... if a power pipeline is used (which isn't a great idea), everything should then be correctly normalised.

Reply:
Though we do ignore this for the pinhole camera, so best to follow the same pattern. i.e assume the sampling is being done through an infinitesimal pinhole.